encord/client.py

```diff
@@ -504,10 +506,32 @@ def add_private_data_to_dataset(
         else:
             raise ValueError(f"Type [{type(private_files)}] of argument private_files is not supported")
 
-        payload = {"files": files, "integration_id": integration_id, "ignore_errors": ignore_errors}
-        response = self._querier.basic_setter(DatasetData, self._config.resource_id, payload=payload)
+        process_hash = self._querier.basic_setter(
+            DatasetDataLongPolling,
+            self._config.resource_id,
+            payload={
+                "files": files,
+                "integration_id": integration_id,
+                "ignore_errors": ignore_errors,
+            },
+        )["process_hash"]
+
+        for _ in range(10):
```

Comment:
How was the 10 chosen in the range? It looks like a short number if we want to have something potentially blocked for hours/days (assuming a reasonable upper-bound timeout of 30s per request @ the LB)

Reply:
switched to while True.
Normally I would not use `while True` instead use `range(n)` but, in this case n could be so big that it works almost like infinity, which is the same as `while True`